pkg/space/space_test.go

```diff
@@ -53,3 +53,30 @@ func TestNewSubSpace(t *testing.T) {
 	)
 	assert.Equal(t, expectedSubSpace, subSpace)
 }
+
+func TestNewSubSubSpace(t *testing.T) {
+	// given
+	srClusterRoles := []string{commoncluster.RoleLabel(commoncluster.Tenant)}
+	sr := spacerequesttest.NewSpaceRequest("jane", "jane-tenant",
+		spacerequesttest.WithTierName("appstudio"),
+		spacerequesttest.WithTargetClusterRoles(srClusterRoles))
+	parentSpace := spacetest.NewSpace(test.HostOperatorNs, "parentSpace")
+	subSpace := NewSubSpace(sr, parentSpace)
+
+	// when
+	subSubSpace := NewSubSpace(sr, subSpace)
+
+	// then
+	expectedSubSubSpace := spacetest.NewSpace(test.HostOperatorNs, SubSpaceName(subSpace, sr),
```

Comment:
based on the changes from the previous comment, let's verify that it's using UID from the right SpaceRequest
```suggestion
	expectedSubSubSpace := spacetest.NewSpace(test.HostOperatorNs, fmt.Sprintf("parentspace-%s", sr.UID[:5]),
```

Reply:
Fixed in 35eb34a.